transmart-core-db/grails-app/services/org/transmartproject/db/clinical/MultidimensionalDataResourceService.groovy

```diff
@@ -314,69 +314,28 @@ class MultidimensionalDataResourceService implements MultiDimensionalDataResourc
         )
     }
 
-    Class aggregateReturnType(AggregateType at) {
+    private static org.hibernate.criterion.Projection projectionForAggregate(AggregateFunction at) {
         switch (at) {
-            case AggregateType.COUNT:
-                return Long
-            case AggregateType.VALUES:
-                return List
-            default:
-                return Number
-        }
-    }
-
-    private String aggregateFieldType(AggregateType at) {
-        switch (at) {
-            case AggregateType.VALUES:
-                return ObservationFact.TYPE_TEXT
-            case AggregateType.COUNT:
-                return null
-            default:
-                return ObservationFact.TYPE_NUMBER
-        }
-    }
-
-    private static org.hibernate.criterion.Projection projectionForAggregate(AggregateType at) {
-        switch (at) {
-            case AggregateType.MIN:
+            case AggregateFunction.MIN:
                 return Projections.min('numberValue')
-            case AggregateType.AVERAGE:
+            case AggregateFunction.AVERAGE:
                 return Projections.avg('numberValue')
-            case AggregateType.MAX:
+            case AggregateFunction.MAX:
                 return Projections.max('numberValue')
-            case AggregateType.COUNT:
+            case AggregateFunction.COUNT:
                 return Projections.rowCount()
-            case AggregateType.PATIENT_COUNT:
+            case AggregateFunction.PATIENT_COUNT:
                 return Projections.countDistinct('patient')
-            case AggregateType.VALUES:
-                return Projections.distinct(Projections.property('textValue'))
+            case AggregateFunction.STD_DEV:
+                return Projections.sqlProjection(
+                        'STDDEV(nval_num) as SD',
```

Comment:
`STDDEV` or `STDDEV_SAMP`? In PostreSQL, stddev is an alias for stddev_samp (https://www.postgresql.org/docs/9.1/static/functions-aggregate.html), in Oracle, the differ (https://docs.oracle.com/cd/B19306_01/server.102/b14200/functions159.htm).

Reply:
Nice catch, thanks! I think we should use `STDDEV_SAMP` then instead.